syntax = "proto3";

package weather;
import "google/protobuf/timestamp.proto";
// Define the weather service
service WeatherService {
  rpc StreamWeather(WeatherRequest) returns (stream WeatherResponse);
}

// Request message (empty)
message WeatherRequest {}

// Response message with weather data
message WeatherResponse {
  string city = 1;
  float temperature = 2;
  int32 humidity = 3;
  string condition = 4;
  google.protobuf.Timestamp timestamp = 5;
}